src/AutomaticVersionRanges.targets

```diff
@@ -0,0 +1,42 @@
+<Project>
+
+  <PropertyGroup>
+    <AutomaticVersionRangesEnabled Condition="'$(AutomaticVersionRangesEnabled)' == '' And '$(Configuration)' == 'Debug'">false</AutomaticVersionRangesEnabled>
+    <AutomaticVersionRangesEnabled Condition="'$(AutomaticVersionRangesEnabled)' == '' And '$(IsPackable)' == 'false'">false</AutomaticVersionRangesEnabled>
+    <AutomaticVersionRangesEnabled Condition="'$(AutomaticVersionRangesEnabled)' == '' And '$(ManagePackageVersionsCentrally)' == 'true'">false</AutomaticVersionRangesEnabled>
+    <AutomaticVersionRangesEnabled Condition="'$(AutomaticVersionRangesEnabled)' == ''">true</AutomaticVersionRangesEnabled>
+  </PropertyGroup>
+
+ <UsingTask TaskName="ConvertToVersionRange" TaskFactory="RoslynCodeTaskFactory" AssemblyFile="$(MSBuildToolsPath)\Microsoft.Build.Tasks.Core.dll" >
+  <Task>
+    <Code Source="$(MSBuildThisFileDirectory)ConvertToVersionRange.cs" />
+  </Task>
+  </UsingTask>
+
+<Target Name="ConvertProjectReferenceVersionsToVersionRanges" AfterTargets="_GetProjectReferenceVersions" Condition="'$(AutomaticVersionRangesEnabled)' == 'true'">
```

Comment:
I am just curious, what is an example of a project reference with a version?

Reply:
All project references have versions added to them internally as part of the packing process. That's how a project reference gets turned into a dependency item on the resulting package.

This target has existed in Particular.Packaging for a long time. Without it, we wouldn't have version ranges on the NServiceBus dependency on testing packages that the core repo produces.

I moved it here for consistency, both in how it's implemented (with the new task) and so that it can more easily be disabled as part of the automatic version range feature.